libzfs.pyx

```diff
@@ -549,11 +549,13 @@ cdef class ZFS(object):
         cdef int prop_id
 
         prop_mapping = {}
+        # If props is None, we include all properties, if it's an empty list, no property is retrieved
+        # except for mountpoint which is separate
         for prop_id in ZFS.proptypes[DatasetType.FILESYSTEM]:
```

Comment:
How about `for prop_id in (ZFS.proptypes[DatasetType.FILESYSTEM] if props is None or len(props) != 0 else []):`

then we avoid even iterating if we dont want any props.